src/Features/Core/Portable/Snippets/SnippetProviders/AbstractInlineStatementSnippetProvider.cs

```diff
@@ -91,6 +113,21 @@ private static bool TryGetInlineExpression(SyntaxToken targetToken, ISyntaxFacts
                 return true;
             }
 
+            expression = null;
+            return false;
+        }
+
+        private static bool TryGetInlineExpressionInQualifiedNameEdgeCases(SyntaxToken targetToken, ISyntaxFactsService syntaxFacts, [NotNullWhen(true)] out SyntaxNode? expression)
+        {
+            var parentNode = targetToken.Parent;
+
+            if (syntaxFacts.IsQualifiedName(parentNode))
```

Comment:
Could you please add comments with more details for the cases this method is trying to handle?

Also, I think it might be better if make the scope of this narrower by explicitly checking for contextual keywords as the RHS of the qualified name

Reply:
> Also, I think it might be better if make the scope of this narrower by explicitly checking for contextual keywords as the RHS of the qualified name

In the issue description I focused on contrextual keywords, but it turned out that the issue is not scoped to them. Consider the following example:
```cs
void M(bool flag)
{
    flag.$$
    Task t;
}
```
Here parser treats `flag.Task` as a qualified name as well, so we need to recover. This PR includes a bunch of test cases where RHS is a simple name, a qualified name on its own or even a generic name (e.g. `InsertInlineSnippetWhenDottingBeforeNameSyntaxTest` for `if` and `while` snippets tests).